pkg/operator/controller.go

```diff
@@ -179,6 +238,13 @@ func (c *Controller) sync() error {
 		klog.Warningf("unknown custom resource state: %s", cr.Spec.ManagementState)
 	}
 
+	// XXX
+	// if hashes are different we need to die.
+	if c.caBundleHash != remoteBundleHash {
+		close(c.stopCh)
+		return nil
+	}
```

Comment:
This should be the first thing done in the sync loop (and put in its own method) so we escape early and leave the prior state intact:

```
func (c *Controller) sync() error {
   caChanged, err := c.syncTrustedCABundle()
   if err != nil {
     return err
   }
  if caChanged {
    c.Stop()
    return nil
  }
  ...
}
```

Reply:
We should also log here that we've detected the CA bundle has changed (and will be shutting down)